sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java

```diff
@@ -38,6 +38,26 @@ private static String getIdMetadataPiece(String[] pieces, int index) {
         return pieces.length >= index + 1 ? pieces[index] : null;
     }
 
+    public static String toHexString(byte[] x509Thumbprint) {
+        if (x509Thumbprint == null) {
+            return null;
+        }
+
+        StringBuilder hexString = new StringBuilder();
+
+        for (byte b : x509Thumbprint) {
+            String hex = Integer.toHexString(0xFF & b);
+
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+
+            hexString.append(hex);
+        }
+
+        return hexString.toString().replace("-", "");
```

Comment:
This is a lot of string manipulation both here and above. And why do you have dashes anyway? I didn't see where those are added above. You might see this in .NET often, but it's bad practice and lazy for .NET, which uses an older function that added them for some reason. I wrote my own for .NET, which is more common these days:

https://github.com/Azure/azure-sdk-for-net/blob/2dc0ca8a6a16025011fca42c671f0052bab922a0/sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Extensions.cs#L16

That said, and as is now the case for newer .NET base64 encoding, it actually uses a preallocated table that is faster. That said, it takes more memory but if used frequently is better. This probably won't get used enough. But, basically, you create a 256-element array with its hex-encoded 2-byte string representation, then just index into that for each byte (as the index, that is).

Reply:
We have `CoreUtils.bytesToHexString` which could replace this entire method